core/core.go

```diff
@@ -580,11 +594,31 @@ func (app *App) Start(ctx context.Context) error {
 		return err
 	}
 
+	// Register and start ordersync service.
+	ordersyncSubprotocols := []ordersync.Subprotocol{
+		NewFilteredPaginationSubprotocol(app, paginationSubprotocolPerPage),
+	}
+	app.ordersyncService = ordersync.New(innerCtx, app.node, ordersyncSubprotocols)
```

Comment:
I think the fact that we are calling `New` in a separate thread than we are calling `GetOrders` makes the case that I describe in the other file (it is the comment that describes an example heuristic) more likely. It's entirely possible that this peer will be hit with a `GetOrders` request from a new peer before it has actually finished bootstrapping it's database using the `ordersync` protocol.

I think that an even more impactful change than the heuristic I described would be for `New` to take a `done` channel that will be sent a signal whenever the `GetOrders` function has finished executing. Prior to receiving the signal, `HandleStream` would reject with a "Database not initialized" warning.

Thoughts?

Reply:
The downside to this approach is that for networks or topics that don't have `minPeers` peers, the ordersync protocol would fail and no one would be able to get any existing orders. I think it's better to leave the implementation as is. It is not very statistically likely that every peer you dial will be "new", especially if we increase `minPeers`.

Reply:
That's a good point. I'm still somewhat concerned about a future in which there are a lot of browser nodes that are spinning up and down quickly and are being used as `ordersync` providers, but I think we can cross that bridge when we come to it considering the downsides that you mention.